rero_ils/modules/jsonresolver.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""General resolver."""
+
+import click
+from flask import current_app
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+
+def resolve_json_refs(pid_type, pid):
+    """Resolver for $ref in record.
+
+    :param pid_type: type of pid
+    :param pid: pid to resolve
+    :return: resolved persistent identifier
+    """
+    try:
+        persistent_id = PersistentIdentifier.get(pid_type, pid)
+    except Exception as err:
+        current_app.logger.error(
+            'Unable to resolve {pid_type} pid: {pid}'.format(
+                pid_type=pid_type,
+                pid=pid
+            )
+        )
+    else:
+        if persistent_id.status == PIDStatus.REGISTERED:
+            return dict(pid=persistent_id.pid_value)
+        current_app.logger.error(
+            '{pid_type} resolve error: /api/documents/{pid} {id}'.format(
+                pid_type=pid_type,
+                pid=pid,
+                id=persistent_id
+            )
```

Comment:
'documents' static string into URL should be dynamic.
Something like
```python
'{pid_type} resolve error: api/{item_route} {id}'.format(
  pid_type=pid_type,
  
item_route=current_app.config.get('RECORDS_REST_ENDPOINTS').get(pid_type).get('item_route').format(pid_value=pid),
  pid=pid,
  id=persistent_id
)
```
I don't know if my `item_route` suggestion works.... not yet the know_how XD